src/traits.rs

```diff
@@ -871,6 +939,22 @@ macro_rules! prim_int_impl {
                 <$T>::rotate_right(self, n)
             }
 
+            fn arithmetic_shl(self, n: u32) -> Self {
+                ((self as $S) << n) as $T
```

Comment:
That is the reason why I would change it's name to `signed` and `unsigned`. It will not confuse users.